rules/DeadCode/Rector/ClassMethod/RemoveParentDelegatingConstructorRector.php

```diff
@@ -230,6 +235,29 @@ private function areConstructorAndParentParameterTypesMatching(
                 if (! $this->nodeComparator->areNodesEqual($parameterType, $parentParameterType)) {
                     return false;
                 }
+
+                if (! $param->default instanceof Expr) {
+                    continue;
+                }
+
+                // native reflection is needed to get exact default value
+                if ($extendedMethodReflection->getDeclaringClass()->getNativeReflection()->hasMethod($methodName)) {
+                    $parentMethod = $extendedMethodReflection->getDeclaringClass()
+                        ->getNativeReflection()
+                        ->getMethod($methodName);
+                    $nativeParentParameterReflection = $parentMethod->getParameters()[$index] ?? null;
+
+                    if (! $nativeParentParameterReflection instanceof ReflectionParameter) {
+                        continue;
+                    }
+
+                    $parentDefault = $nativeParentParameterReflection->getDefaultValue();
+                    $currentDefault = $this->valueResolver->getValue($param->default);
+
+                    if ($parentDefault !== $currentDefault) {
+                        return false;
+                    }
+                }
```

Comment:
Looks good 👍 

Could you extract method here to make code more readable?

Reply:
sure, extracted logic to different method 👍 https://github.com/rectorphp/rector-src/pull/7776/changes/fded30994c63566267acef4b8e18ff8f0302d217